pkg/util/testing/core.go

```diff
@@ -69,6 +70,34 @@ func CheckLatestEvent(ctx context.Context, k8sClient client.Client,
 	return false, fmt.Errorf("mismatch with the latest event: got r:%v t:%v n:%v, reg %v", item.Reason, item.Type, item.Note, item.Regarding)
 }
 
+// CheckEventRecordedAfter checks if an event identified by eventReason, eventType, eventNote was recorded after the provided time t
+func CheckEventRecordedAfter(ctx context.Context, k8sClient client.Client,
```

Comment:
Instead of relying on time, I suggest to have `CountMachingEventRecorded`, then in the tests record the current count, and await for the count to be incremented.

If we want to search for optimizations I would suggest to see if we could narrow down the List by namespace, or the target object.

Reply:
> Instead of relying on time, I suggest to have CountMachingEventRecorded, then in the tests record the current count, and await for the count to be incremented.

Honestly, I'm not sure if this is better than finding by time. I also think we might still encounter flaky tests in this case.

> If we want to search for optimizations I would suggest to see if we could narrow down the List by namespace, or the target object.

Yeah, I think it's make sense.

Reply:
Checking the counter before and after is a common approach, I don't see why this would have a potential to flake, but let me know if I'm missing something.

As for the time-based approach, the potential for flake I see is the optimization based on `lastIndexAfterTime`,  relies on the assumption that the list of events is sorted by time, I'm not sure this is always true. If we drop this check I think we can go ahead also with the approach.

Reply:
> Checking the counter before and after is a common approach, I don't see why this would have a potential to flake, but let me know if I'm missing something.

Ah, I got your point. Yeah, it's make sense.

Reply:
reworked

Reply:
lgtm, left a nit about the error message in the other comment